File: teckwah-TMS/backend/models/UserModel.js
const { DataTypes } = require("sequelize");
const { sequelize } = require("../config/Database");
const bcrypt = require("bcryptjs");

const User = sequelize.define(
  "User",
  {
    user_id: {
      type: DataTypes.STRING(50),
      primaryKey: true,
      allowNull: false,
      comment: "사용자 ID",
    },
    user_password: {
      type: DataTypes.STRING(255),
      allowNull: false,
      comment: "비밀번호 (해시됨)",
    },
    user_department: {
      type: DataTypes.ENUM("CS", "HES", "LENOVO"),
      allowNull: false,
      comment: "사용자 부서",
    },
    user_role: {
      type: DataTypes.ENUM("ADMIN", "USER"),
      allowNull: false,
      comment: "사용자 역할 (ADMIN/USER)",
    },
  },
  {
    tableName: "user",
    timestamps: true,
    underscored: true,
    hooks: {
      beforeCreate: async (user) => {
        if (user.user_password) {
          user.user_password = await bcrypt.hash(user.user_password, 10);
        }
      },
      beforeUpdate: async (user) => {
        if (user.changed("user_password")) {
          user.user_password = await bcrypt.hash(user.user_password, 10);
        }
      },
    },
  }
);

// 비밀번호 검증 메소드
User.prototype.validatePassword = async function (password) {
  return await bcrypt.compare(password, this.user_password);
};

module.exports = User;

File: teckwah-TMS/backend/models/user_model.py
"""
사용자 모델
"""

from sqlalchemy import Column, String, Enum
from sqlalchemy.orm import relationship
from backend.utils.database import Base


class User(Base):
    """사용자 테이블 모델"""
    __tablename__ = "user"

    user_id = Column(String(50), primary_key=True, index=True)
    user_password = Column(String(255), nullable=False)
    user_department = Column(Enum('CS', 'HES', 'LENOVO', name='user_department_enum'), nullable=False)
    user_role = Column(Enum('ADMIN', 'USER', name='user_role_enum'), nullable=False)

    # 관계 설정
    handovers = relationship("Handover", back_populates="user", cascade="all, delete-orphan")
